### ISSUE TITLE
 Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0!

### ISSUE THREAD

**Issue by phoeaven** [Mikubill/sd-webui-controlnet#16]
Errors will be reported when the new version is used:

Error completing request
Arguments: ('task(88y7i4r4jcugm8w)', '1girl', 'lowres, bad anatomy, bad hands, (text:1.6), error, missing fingers, clothe pull, extra digit, fewer digits, (cropped:1.2), (censored:1.2), (low quality, worst quality:1.4), fat, (dress, ribbon:1.2), pubic hair, jpeg artifacts, (signature, watermark, username:1.3), (blurry:1.2), mutated, mutation, out of focus, mutated, extra limb, poorly drawn hands and fingers, missing limb, floating limbs, disconnected limbs, malformed hands and fingers, (motion lines:1.2)', [], 30, 15, False, False, 1, 1, 9, -1.0, -1.0, 0, 0, 0, False, 672, 480, False, 0.35, 2, '4x_foolhardy_Remacri', 20, 0, 0, [], 0, False, False, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'Refresh models', True, 'openpose', 'control_any3_openpose(95070f75)', 1, {'image': array([[[238, 247, 255],
        [236, 245, 254],
        [233, 242, 251],
        ...,
        [192, 203, 225],
        [187, 198, 220],
        [183, 194, 216]],

       [[236, 245, 254],
        [235, 244, 253],
        [233, 242, 251],
        ...,
        [183, 194, 216],
        [178, 189, 211],
        [175, 186, 208]],

       [[233, 242, 251],
        [233, 242, 251],
        [232, 241, 250],
        ...,
        [184, 195, 215],
        [179, 190, 210],
        [176, 187, 207]],

       ...,

       [[181, 181, 189],
        [181, 181, 189],
        [180, 180, 188],
        ...,
        [225, 225, 233],
        [210, 210, 218],
        [201, 201, 209]],

       [[181, 181, 189],
        [181, 181, 189],
        [181, 181, 189],
        ...,
        [224, 224, 232],
        [220, 220, 228],
        [212, 212, 220]],

       [[181, 181, 189],
        [181, 181, 189],
        [181, 181, 189],
        ...,
        [218, 218, 226],
        [225, 225, 233],
        [218, 218, 226]]], dtype=uint8), 'mask': array([[[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       ...,

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]]], dtype=uint8)}, False, 'Scale to Fit (Inner Fit)', True, False, False, 'positive', 'comma', 0, False, False, '', 1, '', 0, '', 0, '', True, False, False, False, 0) {}
Traceback (most recent call last):
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\call_queue.py", line 56, in f
    res = list(func(*args, **kwargs))
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\txt2img.py", line 56, in txt2img
    processed = process_images(p)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\processing.py", line 486, in process_images
    res = process_images_inner(p)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\processing.py", line 628, in process_images_inner
    samples_ddim = p.sample(conditioning=c, unconditional_conditioning=uc, seeds=seeds, subseeds=subseeds, subseed_strength=p.subseed_strength, prompts=prompts)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\processing.py", line 828, in sample
    samples = self.sampler.sample(self, x, conditioning, unconditional_conditioning, image_conditioning=self.txt2img_image_conditioning(x))
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 323, in sample
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args={
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 221, in launch_sampling
    return func()
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 323, in <lambda>
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args={
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\autograd\grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\sampling.py", line 594, in sample_dpmpp_2m
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 135, in forward
    x_out[a:b] = self.inner_model(x_in[a:b], sigma_in[a:b], cond={"c_crossattn": c_crossattn, "c_concat": [image_cond_in[a:b]]})
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\external.py", line 112, in forward
    eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\external.py", line 138, in get_eps
    return self.inner_model.apply_model(*args, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\sd_hijack_utils.py", line 17, in <lambda>
    setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\modules\sd_hijack_utils.py", line 28, in __call__
    return self.__orig_func(*args, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 858, in apply_model
    x_recon = self.model(x_noisy, t, **cond)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1212, in _call_impl
    result = forward_call(*input, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 1329, in forward
    out = self.diffusion_model(x, t, context=cc)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\cldm.py", line 97, in forward2
    return forward(*args, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\cldm.py", line 70, in forward
    emb = self.time_embed(t_emb)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\container.py", line 204, in forward
    input = module(input)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\extensions-builtin\Lora\lora.py", line 178, in lora_Linear_forward
    return lora_forward(self, input, torch.nn.Linear_forward_before_lora(self, input))
  File "C:\Users\60552\Documents\AI folder\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\linear.py", line 114, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument mat1 in method wrapper_addmm)

**Comment by Mikubill** [comment]
Fixed in abb2a08262e1d83a61c8f674e53124b23f5427ac. Update and try again

**Comment by phoeaven** [comment]
> Fixed in [abb2a08](https://github.com/Mikubill/sd-webui-controlnet/commit/abb2a08262e1d83a61c8f674e53124b23f5427ac). Update and try again

Thanks.

**Comment by BlueAccords** [comment]
I got the latest changes but i'm still running into the issue myself.
Using a .safetensors model + control+sd15+openpose file if it makes any difference.

![image](https://user-images.githubusercontent.com/11533167/218639606-7d923066-5a23-4d2e-88b7-0d92a1ba604e.png)

```
Arguments: ('task(l6fzprgqg2m2r59)', 'masterpiece, best quality, ultra-detailed, illustration, 1girl, white hair, golden eyes, long hair, halo, angel wings, serene expression', 'lowres, ((bad anatomy)), ((bad hands)), text, missing finger, extra digits, fewer digits, blurry, ((mutated hands and fingers)), (poorly drawn face), ((mutation)), ((deformed face)), (ugly), ((bad proportions)), ((extra limbs)), extra face, (double head), (extra head), ((extra feet)), monster, logo, cropped, worst quality, low quality, normal quality, jpeg, humpbacked, long body, long neck, ((jpeg artifacts))', [], 30, 0, False, False, 1, 1, 8, -1.0, -1.0, 0, 0, 0, False, 512, 512, False, 0.7, 2, 'Latent', 0, 0, 0, [], 0, False, False, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'Refresh models', True, 'openpose', 'control_sd15_openpose(fef5e48e)', 1, {'image': array([[[ 0,  1,  7],
        [ 0,  0,  7],
        [ 0,  1,  8],
        ...,
        [ 0,  0,  3],
        [ 0,  0,  3],
        [ 0,  0,  2]],

       [[ 0,  3, 11],
        [ 0,  0,  8],
        [ 0,  0,  8],
        ...,
        [ 0,  6, 11],
        [ 0,  8, 11],
        [ 0,  8, 11]],

       [[11, 27, 36],
        [ 4, 20, 29],
        [ 2, 16, 25],
        ...,
        [25, 37, 43],
        [26, 40, 45],
        [26, 40, 45]],

       ...,

       [[ 0,  0,  2],
        [ 0,  0,  2],
        [ 0,  0,  2],
        ...,
        [ 0,  0,  0],
        [ 0,  0,  0],
        [ 0,  0,  0]],

       [[ 0,  0,  2],
        [ 0,  0,  2],
        [ 0,  0,  2],
        ...,
        [ 0,  0,  0],
        [ 0,  0,  0],
        [ 0,  0,  0]],

       [[ 0,  0,  2],
        [ 0,  0,  2],
        [ 0,  0,  2],
        ...,
        [ 0,  0,  0],
        [ 0,  0,  0],
        [ 0,  0,  0]]], dtype=uint8), 'mask': array([[[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       ...,

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]],

       [[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]]], dtype=uint8)}, False, 'Scale to Fit (Inner Fit)', False, False, False, 'positive', 'comma', 0, False, False, '', False, 1, '', 0, '', 0, '', True, False, False, False, 0) {}
Traceback (most recent call last):
  File "A:\stablediff\stable-diffusion-webui\modules\call_queue.py", line 56, in f
    res = list(func(*args, **kwargs))
  File "A:\stablediff\stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\modules\txt2img.py", line 56, in txt2img
    processed = process_images(p)
  File "A:\stablediff\stable-diffusion-webui\modules\processing.py", line 486, in process_images
    res = process_images_inner(p)
  File "A:\stablediff\stable-diffusion-webui\modules\processing.py", line 628, in process_images_inner
    samples_ddim = p.sample(conditioning=c, unconditional_conditioning=uc, seeds=seeds, subseeds=subseeds, subseed_strength=p.subseed_strength, prompts=prompts)
  File "A:\stablediff\stable-diffusion-webui\modules\processing.py", line 828, in sample
    samples = self.sampler.sample(self, x, conditioning, unconditional_conditioning, image_conditioning=self.txt2img_image_conditioning(x))
  File "A:\stablediff\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 323, in sample
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args={
  File "A:\stablediff\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 221, in launch_sampling
    return func()
  File "A:\stablediff\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 323, in <lambda>
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args={
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\autograd\grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\sampling.py", line 145, in sample_euler_ancestral
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 135, in forward
    x_out[a:b] = self.inner_model(x_in[a:b], sigma_in[a:b], cond={"c_crossattn": c_crossattn, "c_concat": [image_cond_in[a:b]]})
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\external.py", line 112, in forward
    eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
  File "A:\stablediff\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\external.py", line 138, in get_eps
    return self.inner_model.apply_model(*args, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\modules\sd_hijack_utils.py", line 17, in <lambda>
    setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
  File "A:\stablediff\stable-diffusion-webui\modules\sd_hijack_utils.py", line 28, in __call__
    return self.__orig_func(*args, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 858, in apply_model
    x_recon = self.model(x_noisy, t, **cond)
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1148, in _call_impl
    result = forward_call(*input, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 1329, in forward
    out = self.diffusion_model(x, t, context=cc)
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\cldm.py", line 97, in forward2
    return forward(*args, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\cldm.py", line 70, in forward
    emb = self.time_embed(t_emb)
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\container.py", line 139, in forward    input = module(input)
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "A:\stablediff\stable-diffusion-webui\extensions-builtin\Lora\lora.py", line 178, in lora_Linear_forward
    return lora_forward(self, input, torch.nn.Linear_forward_before_lora(self, input))
  File "A:\stablediff\stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\linear.py", line 114, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument mat1 in method wrapper_addmm)
```
